src/compiler/checker.ts

```diff
@@ -4479,7 +4499,10 @@ namespace ts {
 
         function getPropertiesOfType(type: Type): Symbol[] {
             type = getApparentType(type);
-            return type.flags & TypeFlags.UnionOrIntersection ? getPropertiesOfUnionOrIntersectionType(<UnionType>type) : getPropertiesOfObjectType(type);
+            if (type.flags & TypeFlags.UnionOrIntersection) {
+                return getPropertiesOfUnionOrIntersectionType(<UnionType>type);
+            }
+            return getPropertiesOfObjectType(type);

```

Comment:
With a separate spread type, I had more code in here that made the verbose style more readable. Restored.